cargo test meet two error

**Version**
monoio v0.0.3

**Platform**
Linux n227-074-054 4.14.81.bm.15-amd64 #1 SMP Debian 4.14.81.bm.15 Sun Sep 8 05:02:31 UTC 2019 x86_64 GNU/Linux

**Description**
when i run cargo test, there are two errors.

```
➜  monoio git:(master) cargo test
   Compiling monoio v0.0.3 (/data00/home/yuyanglong/code/rust/yuyanglong/other-projects/monoio/monoio)
error[E0271]: type mismatch resolving `<impl futures::Future<Output = [async output]> as futures::Future>::Output == ()`
  --> monoio/src/stream/stream_ext.rs:44:30
   |
34 |           async move {
   |  ____________________-
35 | |             while let Some(item) = self.next().await {
36 | |                 (f)(item);
37 | |             }
38 | |         }
   | |_________- the found `async` block
...
44 |   type ForEachFut<T, Fut, F> = impl Future<Output = ()>;
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found associated type
   |
  ::: /data00/home/yuyanglong/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:65:43
   |
65 |   pub const fn from_generator<T>(gen: T) -> impl Future<Output = T::Return>
   |                                             ------------------------------- the found opaque type
   |
   = note:    expected unit type `()`
           found associated type `<impl futures::Future<Output = [async output]> as futures::Future>::Output`
   = help: consider constraining the associated type `<impl futures::Future<Output = [async output]> as futures::Future>::Output` to `()`
   = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html

error[E0277]: the trait bound `T: stream::Stream` is not satisfied
  --> monoio/src/stream/stream_ext.rs:44:30
   |
44 | type ForEachFut<T, Fut, F> = impl Future<Output = ()>;
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `stream::Stream` is not implemented for `T`
   |
help: consider restricting type parameter `T`
   |
44 | type ForEachFut<T: stream::Stream, Fut, F> = impl Future<Output = ()>;
   |                  ++++++++++++++++

Some errors have detailed explanations: E0271, E0277.
For more information about an error, try `rustc --explain E0271`.
error: could not compile `monoio` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/deny.toml
+++ b/deny.toml
@@ -5,6 +5,7 @@ allow = [
     "Apache-2.0",
     "Apache-2.0 WITH LLVM-exception",
     "Zlib",
+    "BSD-3-Clause",
     "Unlicense"
 ]
 copyleft = "deny"

--- a/monoio/src/stream/stream_ext.rs
+++ b/monoio/src/stream/stream_ext.rs
@@ -33,7 +33,7 @@ pub trait StreamExt: Stream {
     {
         async move {
             while let Some(item) = self.next().await {
-                (f)(item);
+                (f)(item).await;
             }
         }
     }
@@ -41,7 +41,7 @@ pub trait StreamExt: Stream {
 
 impl<T> StreamExt for T where T: Stream {}
 
-type ForEachFut<T, Fut, F> = impl Future<Output = ()>;
+type ForEachFut<T: Stream, Fut, F> = impl Future<Output = ()>;
 
 #[must_use = "streams do nothing unless polled"]
 pub struct Map<St, F> {

